OpenEdXMobile/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -183,6 +189,62 @@ public void onClick(View v) {
         tryToSetUIInteraction(true);
     }
 
+    private void validateRegistrationFields() {
+        Bundle parameters = getRegistrationParameter();
+        if (parameters == null) {
+            return;
+        }
+        showProgress();
+        final Map<String, String> parameterMap = new HashMap<>();
+        for (String key : parameters.keySet()) {
+            parameterMap.put(key, parameters.getString(key));
+        }
+        Call<JsonObject> validateRegistrationFields = loginService.validateRegistrationFields(parameterMap);
+        validateRegistrationFields.enqueue(new ErrorHandlingCallback<JsonObject>(this) {
+            @Override
+            protected void onResponse(@NonNull JsonObject responseBody) {
+                final Type stringMapType = new TypeToken<HashMap<String, String>>() {
+                }.getType();
+                final HashMap<String, String> messageBody = new Gson().fromJson(responseBody.get("validation_decisions"), stringMapType);
+                if (hasValidationError(messageBody)) {
+                    hideProgress();
+                } else {
+                    createAccount(parameters);
+                }
+            }
+
+            @Override
+            protected void onFailure(@NonNull Throwable error) {
+                hideProgress();
+                RegisterActivity.this.showAlertDialog(null, ErrorUtils.getErrorMessage(error,
+                        RegisterActivity.this));
+                logger.error(error);
+            }
+        });
+    }
+
+    private boolean hasValidationError(HashMap<String, String> messageBody) {
+        boolean errorShown = false;
+        for (String key : messageBody.keySet()) {
```

Comment:
Can we make a common function for this as [L:329](https://github.com/edx/edx-app-android/blob/4ee2b7f7312c1535e3c07be4f9a5d7877db900ae/OpenEdXMobile/src/main/java/org/edx/mobile/view/RegisterActivity.java#L329) has the same code

Reply:
I already tried but both lines of codes have the following differences. 
- Using Difference HashMap types.
- The availability of error messages are different.
- Scenarios Conditions.

Also, the effort is more than it's worth to create a common method.